src/main/java/randoop/generation/DemandDrivenInputCreator.java

```diff
@@ -0,0 +1,406 @@
+import org.checkerframework.checker.nullness.qual.Nullable;
+import randoop.DummyVisitor;
+import randoop.ExecutionOutcome;
+import randoop.NormalExecution;
+import randoop.operation.CallableOperation;
+import randoop.operation.ConstructorCall;
+import randoop.operation.MethodCall;
+import randoop.operation.TypedClassOperation;
+import randoop.operation.TypedOperation;
+import randoop.sequence.ExecutableSequence;
+import randoop.sequence.Sequence;
+import randoop.sequence.SequenceCollection;
+import randoop.test.DummyCheckGenerator;
+import randoop.types.ArrayType;
+import randoop.types.NonParameterizedType;
+import randoop.types.Type;
+import randoop.types.TypeTuple;
+import randoop.util.EquivalenceChecker;
+import randoop.util.Log;
+import randoop.util.Randomness;
+import randoop.util.SimpleArrayList;
+import randoop.util.SimpleList;
+
+/**
+ * A demand-driven approach to construct inputs. Randoop works by selecting a method and then trying
+ * to find inputs for that method. Ordinarily, Randoop works bottom-up: if Randoop cannot find
+ * inputs for the selected method, it gives up and selects a different method. This demand-driven
+ * approach works top-down: if Randoop cannot find inputs for the selected method, then it looks for
+ * methods that create values of the necessary type, and iteratively tries to call them.
+ *
+ * <p>The demand-driven approach implements the "Detective" component described by the ASE 2015
+ * paper <a href="https://people.kth.se/~artho/papers/lei-ase2015.pdf">"GRT: Program-Analysis-Guided
+ * Random Testing" by Ma et al.</a> .
+ */
+public class DemandDrivenInputCreator {
+  /**
+   * The principal set of sequences used to create other, larger sequences by the generator. New
+   * sequences are added on demand for creating object of missing types. Shared with {@link
+   * ComponentManager#gralComponents}.
+   */
+  private final SequenceCollection sequenceCollection;
+
+  /**
+   * A secondary sequence collection used to store sequences that are generated during the
+   * demand-driven input creation process. These sequences are added to the main sequence collection
+   * after the demand-driven input creation process is complete.
+   *
+   * <p>This is an optimization to reduce the search space for the missing types in the main
+   * sequence collection.
+   */
+  private final SequenceCollection secondarySequenceCollection;
+
+  /**
+   * If true, {@link #createInputForType(Type)} returns only sequences that declare values of the
+   * exact type that was requested.
+   */
+  private boolean exactTypeMatch;
+
+  /**
+   * If true, {@link #createInputForType(Type)} returns only sequences that are appropriate to use
+   * as a method call receiver, i.e., Type.isNonreceiverType() returns false for the type of the
+   * variable created by the sequence.
+   */
+  private boolean onlyReceivers;
+
+  /** Constructs a new {@code DemandDrivenInputCreator} object. */
+  public DemandDrivenInputCreator(
```

Comment:
Please document the meaning of each formal parameter.  For example, is `sequenceCollection` Randoop's full pool, or is it something that is small and created by the demand-driven algoritm?